### ISSUE TITLE
[bug] Showing window via system tray menu item requires an extra click to focus

### ISSUE THREAD

**Issue by egoist** [tauri-apps/tauri#7884]
### Describe the bug

I added a system tray menu item to manually show the main window, but it can't be focused without an extra click on the window, open the window via left-clicking the tray icon works fine.

### Reproduction

here's the repro repository: https://github.com/egoist-bot/tauri-bug (generated from create-tauri-app, with changes to add system tray you can see here https://github.com/egoist-bot/tauri-bug/commit/192baa0a1583ae5d393cbc33209a2d118e4ede65)

```bash
git clone https://github.com/egoist-bot/tauri-bug.git
cd tauri-bug
pnpm i
pnpm tauri dev
```

When the app is ready, right click the menubar icon and click "Show".

You'll notice the window is visible but not focused, it requires an extra click on the window to focus.

https://github.com/tauri-apps/tauri/assets/8784712/c232c1db-7990-482d-b000-7f2af79bd859

### Expected behavior

it should focus without an extra click

### Platform and versions

```text
[✔] Environment
    - OS: Mac OS 14.0.0 X64
    ✔ Xcode Command Line Tools: installed
    ✔ rustc: 1.70.0 (90c541806 2023-05-31)
    ✔ Cargo: 1.70.0 (ec8a8a0ca 2023-04-25)
    ✔ rustup: 1.26.0 (5af9b9484 2023-04-05)
    ✔ Rust toolchain: stable-aarch64-apple-darwin (default)
    - node: 18.16.0
    - pnpm: 8.6.11
    - yarn: 1.22.19
    - npm: 9.5.1

[-] Packages
    - tauri [RUST]: 1.4.1
    - tauri-build [RUST]: 1.4.0
    - wry [RUST]: 0.24.4
    - tao [RUST]: 0.16.2
    - @tauri-apps/api [NPM]: 1.4.0
    - @tauri-apps/cli [NPM]: 1.4.0

[-] App
    - build-type: bundle
    - CSP: unset
    - distDir: ../dist
    - devPath: http://localhost:1420/
    - framework: React
    - bundler: Vite
```

**Comment by amrbashir** [comment]
I see in the repro you also have it show when your left click the tray icon itself, does that work fine or shows the same behavior?

Also does it make any difference if you call `app.show()` before setting focus to the window?

**Comment by egoist** [comment]
@amrbashir  left click the tray icon works fine.

**Comment by egoist** [comment]
@amrbashir and calling `app.show()` makes no difference

**Comment by amrbashir** [comment]
thanks for the info, cc @pewsheen

**Comment by pewsheen** [comment]
Calling `[button performClick:nil]` in the custom tray class's `click` selector somehow conflicts with the focus

This issue should be fixed in Tauri v2 with [refactored tray-icon implementation](https://github.com/tauri-apps/tray-icon/commit/54fc7de37c3568312b27c30bdd22e830b1f15a3b).

**Comment by amrbashir** [comment]
Could we backport this fix to tao 0.16 if possible so it could be published in an upcoming tauri release?

**Comment by amrbashir** [comment]
tao v0.16.3 should be released soon with this fix, run `cargo update` inside `src-tauri`. If the issue persists, please let us know
